pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -223,6 +225,9 @@ type Controller struct {
 	// if set to true, NodeController will taint Nodes based on its condition for 'NetworkUnavailable',
 	// 'MemoryPressure', 'OutOfDisk' and 'DiskPressure'.
 	taintNodeByCondition bool
+
+	nodeUpdateChannels []chan *v1.Node
+	nodeUpdateQueue    *cache.FIFO
```

Comment:
Instead of using cache.FIFO, can you use workqueue.Interface which was designed exactly for this specific usecase (handling duplicates, errors, etc.)